### PR TITLE
Add Synthetic CrossCat datasets

### PR DESCRIPTION
**Pull request by srvasude**
Add four synthetic CrossCat datasets for each of the data types. Will add unit tests that verify that we get the appropriate clusterings with these later.

### CODE REVIEW

File: cxx/assets/test_files/test_crosscat_bernoulli.schema [hunk 1]
```diff
@@ -0,0 +1,4 @@
+col1 ~ bernoulli(id)
```
Comment (ThomasColthurst) [inline cxx/assets/test_files/test_crosscat_bernoulli.schema:1]: Doesn't the relation name here have to match the second column of the .obs file?  Here it is "col1" but the .obs file uses "has_col1".

File: cxx/assets/test_files/test_crosscat_categorical.schema [hunk 2]
```diff
@@ -0,0 +1,4 @@
+col1 ~ stringcat[strings="a:b:c:d",delim=:](id)
```
Comment (ThomasColthurst) [inline cxx/assets/test_files/test_crosscat_categorical.schema:1]: In addition to the "has_" issue, there also appears to be an off by one error.  Here, "col1" is the one that has a/b/c/d values, but it is "has_col0" in the .obs file that has those values.  

Similarly for the other relations.

File: cxx/assets/test_files/test_crosscat_generator.py [hunk 3]
```diff
@@ -0,0 +1,87 @@
+# Generates a 4 column dataset for one data type.
+import numpy as np
+
+NUM_SAMPLES_1 = 33
+NUM_SAMPLES_2 = 50
+NUM_SAMPLES = 100
+
```
Comment (ThomasColthurst) [inline cxx/assets/test_files/test_crosscat_generator.py:7]: What do you think about adding some tests for this program?

File: cxx/assets/test_files/test_crosscat_bernoulli.schema (hunk 1 above)
Comment (srvasude) [reply cxx/assets/test_files/test_crosscat_bernoulli.schema:1]: Yeah this was a bad push. Fixed here and elsewhere.

File: cxx/assets/test_files/test_crosscat_categorical.schema (hunk 2 above)
Comment (srvasude) [reply cxx/assets/test_files/test_crosscat_categorical.schema:1]: Fixed as well.

File: cxx/assets/test_files/test_crosscat_generator.py (hunk 3 above)
Comment (srvasude) [reply cxx/assets/test_files/test_crosscat_generator.py:7]: I'm adding integration tests in a later PR that will test the files (test some basic invariants, like creating two IRMs, etc)

**Review by ThomasColthurst — APPROVED** [review]

**Pull request merged**